### PR TITLE
Health: Move all crates to `crates/`

### PR DESCRIPTION
**Pull request by johnyob**
# Context

**Related Tasks**: [Use `/crates` for crates directory](https://app.asana.com/0/1205770721173531/1205870630092075/f)

To conform to existing standards in larger Rust projects, we will be moving all crates to `crates/`. 
There is already precedence for this style of folder separation in `jstz` e.g TypeScript packages in `packages/`.

Additionally this clears the root folder (which is slowly becoming more cluttered 😬 )

# Description

This PR moves all crates to `crates/`. Additionally to fit better with smart contract practices, all contracts (e.g. from `jstz_bridge`) have been moved to `contracts/`. 

All commit history is retained using `git mv` (although this requires a `--follow` flag when using `git log` manually). 

# Manually testing the PR

None required :)

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -6,18 +6,6 @@ updates:
       interval: daily
     commit-message:
       prefix: "Deps (NPM)"
-  - package-ecosystem: npm
```
Comment (alanmarkoTrilitech) [inline .github/dependabot.yml:9]: Why are these being removed? Shouldn't there be new settings for new directories?

**Review by alanmarkoTrilitech — APPROVED** [review]
LGTM, but please address the question regarding the dependabot changes before merging.

File: .github/dependabot.yml (hunk 1 above)
Comment (johnyob) [reply .github/dependabot.yml:9]: Dependabot should be able to traverse our file structure to find relevant manifest files for NPM (`package.json`), Cargo (`Cargo.toml`) and GitHub Actions (.yml file in `.github/workflows`)

Comment (johnyob) [reply .github/dependabot.yml:9]: Actually I'm mistaken, only the nightly version of dependabot does this 😔 I will update the config to have all the relevant paths

Comment (johnyob) [reply .github/dependabot.yml:9]: @alanmarkoTrilitech updated

**Review by alanmarkoTrilitech — APPROVED** [review]

**Pull request merged**